src/main/java/com/google/starfish/servlets/SearchServlet.java

```diff
@@ -0,0 +1,94 @@
+package com.google.starfish.servlets;  
+
+import java.io.IOException;  
+import java.sql.Connection;  
+import java.sql.SQLException;  
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import javax.servlet.ServletException;  
+import javax.servlet.annotation.WebServlet;  
+import javax.servlet.http.HttpServlet;  
+import javax.servlet.http.HttpServletRequest;  
+import javax.servlet.http.HttpServletResponse;  
+import javax.sql.DataSource;
+import java.util.Date;
+import com.google.starfish.services.NoteService;
+import com.google.starfish.models.Note;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+
+/** Servlet that returns search results for notes based on school and course */
+@WebServlet("/search")  
+public class SearchServlet extends HttpServlet {  
+
+  private NoteService noteService = new NoteService();
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");  
+    String reqSchool = req.getParameter("school").toLowerCase().trim();
```

Comment:
I haven't yet seen where this request is sent, and perhaps you have a safeguard there to make sure "school" and "course" are sent?  But in case you don't, or maybe even if you do (since theoretically this servlet could be reusable in different contexts where, you may forget to require it or so), you might consider extracting this out into a helper method where you verify first that the parameter exists, and then proceed to make it lower case and trim

Reply:
This is a really good point, the way it is right now if school or course were null, then we'd get an exception when we try toLowerCase() and trim() it. I think a helper method that does a null check and responds accordingly is the best move.